config/hwloc.m4

```diff
@@ -1019,6 +1019,39 @@ return clGetDeviceIDs(0, 0, 0, NULL, NULL);
     fi
     # don't add LIBS/CFLAGS/REQUIRES yet, depends on plugins
 
+    # RSMI support
+    hwloc_rsmi_happy=no
+    if test "x$enable_io" != xno && test "x$enable_rsmi" != "xno"; then
+      hwloc_rsmi_happy=yes
+      AC_CHECK_HEADERS([rocm_smi.h], [
+        AC_CHECK_LIB([rocm_smi64], [rsmi_init], [HWLOC_RSMI_LIBS="-lrocm_smi64"], [hwloc_rsmi_happy=no])
+        ], [hwloc_rsmi_happy=no])
+    fi
+    if test "x$hwloc_rsmi_happy" = "xyes"; then
+      tmp_save_CFLAGS="$CFLAGS"
+      CFLAGS="$CFLAGS $HWLOC_RSMI_CFLAGS"
+      tmp_save_LIBS="$LIBS"
+      LIBS="$LIBS $HWLOC_RSMI_LIBS"
+      # AC_CHECK_DECLS([nvmlDeviceGetMaxPcieLinkGeneration],,[:],[[#include <nvml.h>]])
+      # CFLAGS="$tmp_save_CFLAGS"
+      # LIBS="$tmp_save_LIBS"
+    fi
```

Comment:
If you don't have any specific function to check (something that was added after the first RSMI release and that you want to put under #ifdef in hwloc), remove this entire "if/fi" block.

Reply:
will remove it.